votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,262 @@
+        self.storage.replace(storage.unwrap_import().into_remote());
+    }
+}
+
+impl InitGenesis for Context {
+    fn begin_genesis(&mut self) {}
+
+    fn init_genesis(&mut self, config: &[u8]) {
+    }
+
+    fn end_genesis(&mut self) {}
+}
+
+
+impl TxOwner for Context {
+    fn block_opened(&mut self, _: &Header) -> Result<(), HeaderError> {
+        Ok(())
+    }
+
+    fn execute_transaction(&mut self, transaction: &Transaction) -> Result<TransactionOutcome, ()> {
+        if let Err(error) = self.excute_tx(transaction) {
+            match error {
+                ExecuteError::InvalidMetadata => Err(()),
+                ExecuteError::InvalidSign => Err(()),
+                ExecuteError::InvalidFormat => Err(()),
+            }
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn check_transaction(&self, transaction: &Transaction) -> Result<(), coordinator::types::ErrorCode> {
+        let todo_fixthis: coordinator::types::ErrorCode = 3;
+        assert_eq!(transaction.tx_type(), "CreateGeneralMeeting");
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| todo_fixthis)?;
+        tx.verify().map_err(|_| todo_fixthis)?;
+        Ok(())
+    }
+
+    fn block_closed(&mut self) -> Result<Vec<Event>, CloseBlockError> {
+        Ok(Vec::new())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct TimeStamp{
+    time: u64,
+}
+
+
+impl TimeStamp {
+    fn set_time(&mut self, time: u64){
+        self.time = time;
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct GeneralMeetingId {
+    id: u8,
+}
+
+impl GeneralMeetingId {
+    pub fn new ()->Self {
+        let mut rng = rand::thread_rng();
+        let random_id = rng.gen();
+        Self{
+            id: random_id
+        }
+    }
+}
+
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct VoteResult{
+    favor:u32,
+    against:u32,
+    absention: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum GeneralMeetingError {
+    NoSuchMeeting,
+    MeetingExist,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct GeneralMeeting {
+    id: GeneralMeetingId,
+    number_of_options: u32,
+    voting_end_time: TimeStamp,
+    tallying_time: TimeStamp,
+    result: Option<Vec<VoteResult>>,
+}
+
+impl GeneralMeeting {
+    pub fn new(end_time: TimeStamp, tallying_time: TimeStamp) -> Self {
+        let meeting_id= GeneralMeetingId::new();
+        let options = 4;
+        let final_result = Some(Vec::new());
+        Self{
+            id: meeting_id,
+            number_of_options: options,
+            voting_end_time: end_time,
+            tallying_time,
+            result: final_result
+        }
+    }
+}
+
+#[service]
+pub trait MeetingManager: Service {
+    fn create_general_meeting(&mut self, general_meeting: GeneralMeeting, id: GeneralMeetingId) -> Result<GeneralMeetingId,GeneralMeetingError>;
+}
```

Comment:
Why did you create `MeetingManager` service?
Which module will use this service?